lib/checkbufferoverrun.cpp

```diff
@@ -101,7 +101,7 @@ void CheckBufferOverrun::arrayIndexOutOfBoundsError(const Token *tok, const Arra
                     continue;
                 std::string nr;
                 if (index.size() > 1U)
-                    nr = "(" + MathLib::toString(i + 1) + getOrdinalText(i + 1) + " array index) ";
+                    nr = "(" + MathLib::toString(i + static_cast<std::size_t>(1u)) + getOrdinalText(static_cast<int>(i) + 1) + " array index) ";
```

Comment:
I don't think this fix looks good. unless you tell me that there is a bug here I don't see I don't plan to accept this.